File: src/java/cl/unab/cliquecover/InstanceReader.java
package cl.unab.cliquecover;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class InstanceReader {

    public static Grafo leerArchivo(String rutaArchivo) throws IOException {
        try (BufferedReader br = new BufferedReader(new FileReader(rutaArchivo))) {
            String linea;
            
            do {
                linea = br.readLine();
            } while (linea != null && linea.trim().startsWith("#"));

            if (linea == null) {
                throw new IllegalArgumentException("El archivo está vacío o mal formateado.");
            }

            String[] partes = linea.trim().split("\\s+");
            int v = Integer.parseInt(partes[0]);
            int e = Integer.parseInt(partes[1]);

            Grafo grafo = new Grafo(v);

            int aristasLeidas = 0;
            while ((linea = br.readLine()) != null) {
                linea = linea.trim();
                if (linea.isEmpty() || linea.startsWith("#")) continue;

                partes = linea.split("\\s+");
                int u = Integer.parseInt(partes[0]);
                int w = Integer.parseInt(partes[1]);
                
                grafo.agregarArista(u, w);
                aristasLeidas++;
            }

            return grafo;
        }
    }
}

File: src/python/instance_reader.py
from grafo import Grafo

def leer_archivo(ruta_archivo):
    try:
        with open(ruta_archivo, 'r') as file:
            lineas = file.readlines()

        grafo = None
        for linea in lineas:
            linea = linea.strip()
            if not linea or linea.startswith("#"):
                continue

            partes = linea.split()

            if grafo is None:
                v = int(partes[0])
                grafo = Grafo(v)
            else:
                u, w = int(partes[0]), int(partes[1])
                grafo.agregar_arista(u, w)

        return grafo

    except Exception as e:
        print(f"Error al leer el archivo: {e}")
        return None